python/ray/util/client/dataclient.py

```diff
@@ -35,6 +34,11 @@ def __init__(self, channel: "grpc._channel.Channel", client_id: str,
         self.ready_data: Dict[int, Any] = {}
         self.cv = threading.Condition()
         self.lock = threading.RLock()
+
+        # NOTE: Dictionary insertion is guaranteed to complete before lookup
+        # and/or removal because of synchronization via the request_queue.
+        self.asyncio_waiting_data: Dict[int, Callable[
+            [ray_client_pb2.DataResponse], None]] = {}
```

Comment:
should we give Dict[int, Callable[[ray_client_pb2.DataResponse], None]] a type name at the top of the file?

Reply:
I'll give the callback a typename

Reply:
Great idea!